### PR TITLE
fix(tutorials): solving ux issues within tutorial modal

### PR DESCRIPTION
**Pull request by gratcliff**
## 🧰 What's being changed?

- Moving tutorialModal to top level and adding default object to maintain the modal in the dom
- Fixing issue where steps were not reset on reopen due to reference error
- Updating @readme/ui
- Updating tutorial demo

## 🧪 Testing

Enable tutorials via toggle.
- Click out of modal via "x" (css murky here, blacked out) - should remove backdrop on close
- Click into tutorial seconds step, close modal, and reopen - should open on first step.

### CODE REVIEW

**Review by rafegoldberg — COMMENTED** [review]
I ought to check this out locally before I approve, but I read through the code and the changes look good! (Speaking of, we should _really_ set up PR apps for this repo again. I still don't understand why the Heroku pipeline I created for this months and months ago just up and disappeared, but I do remember it being quite easy to set up...)

File: packages/api-explorer/src/index.jsx [hunk 1]
```diff
@@ -1,18 +1,48 @@
...
 
 const ErrorBoundary = require('./ErrorBoundary');
 const Doc = require('./Doc');
+const { TutorialModal } = require('@readme/ui/.bundles/es/ui/compositions');
 
 const getAuth = require('./lib/get-auth');
 
 const supportedHttpMethods = ['connect', 'delete', 'get', 'head', 'options', 'patch', 'post', 'put', 'trace'];
 
+const defaultTutorial = {
```
Comment (erunion) [inline packages/api-explorer/src/index.jsx:18]: Can this live in the `TutorialModal` instead as a default prop? The explorer having to know the shape of a tutorial here seems out of scope for the explorer.

Comment (gratcliff) [reply packages/api-explorer/src/index.jsx]: Yeah good call, I forgot we had it defined there (although a v long path). Fixed this.

**Comment by erunion** [comment]
@rafegoldberg I'm going to put something on our pod board to get PR apps working again here.

**Review by erunion — APPROVED** [review]
